public/scripts/editor/js/bramble-editor.js

```diff
@@ -35,12 +36,18 @@ define(function(require) {
       });
 
       // Start loading the project files
-      Project.load(makeDetails, host, authenticated, function(err, fileToOpen) {
+      Project.load(projectDetails, host, function(err, fileToOpen) {
         if(err) {
           console.error("[Thimble Error]", err);
           return;
         }
 
+        // Initialize the project name UI
+        projectNameComponent = new ProjectRenameUtility(host, authenticated, csrfToken);
```

Comment:
Let's not block on the project loading because that holds up the UI. Instead, do this:
In `main.js` we initialize the project because the context should be set **before** loading the editor viz. bramble-editor.js:
```js
var Project = require("project");

// Here `Project.init` merely sets the values of the project i.e. line 83-89 in `project.js`
Project.init(projectDetails);

// This should not be in `bramble-editor.js` because the editor has nothing to do with login
SSOOverride.init();
```

Then in `bramble-editor.js`, move the `projectNameComponent` assignment outside `Project.load`. It might even make sense to move the `projectNameComponent` stuff into the `Project.init` code.